### ISSUE TITLE
This mods looks awesome! Any chance of a datapack implementation?

### ISSUE THREAD

**Issue by Yarden-zamir** [SamsTheNerd/wnboi#1]
This feels like an amazing resource for modpacks and I'd love to use it, would it be possible to extend this a little to allow for datapack defined wnbois?

**Comment by SamsTheNerd** [comment]
That certainly sounds interesting ! I haven't really used data packs outside of recipes and tags, could you describe an example use case for it ?

**Comment by Yarden-zamir** [comment]
Example of how diesel generators mod adds new fuels
`data/createdieselgenerators/diesel_engine_fuel_types/milk.json`
```json
{
  "fluid": "#forge:milk",
  "sound_speed": 8,
  "normal": {
    "speed": 16.0,
    "strength": 1024.0,
    "burn_rate": 2
  },
  "modular": {
    "speed": 16.0,
    "strength": 1024.0,
    "burn_rate": 2
  },
  "huge": {
    "speed": 8.0,
    "strength": 1024.0,
    "burn_rate": 3
  }
}
```

example of how I can imagine your mod would work
`data/wnboi/varients/plank_based_blocks.json`
```json
{
	"entries":[
		"minecraft:oak_planks",
		"minecraft:oak_stairs",
		"minecraft:oak_fence"
	]
}
```

**Comment by Yarden-zamir** [comment]
This also means that other mods can add that sort of thing into their data folder and they will automatically get compatibility without hooking into any code

**Comment by SamsTheNerd** [comment]
Sorry I'm not sure I follow, what would this actually do in game ? 

I was thinking the wheel sections could be linked to data pack functions and then link the wheel to some item to use it with.

**Comment by Yarden-zamir** [comment]
My thought is that I use the wheel to cycle between items that are in the same list
Datapack functions sound fun as well though.

I'm a part of the Modpack team Prepare to Dye and we have a system of devices which are categories of same type machines that can be converted between eachother on the fly. Currently we are using a keybind to open the stonecutter interface and use that to cycle. Your mod looks perfect for us for this usecase but I see that the way to define those currently is only through a mod addon

**Comment by Yarden-zamir** [comment]
Does that make sense?

**Comment by SamsTheNerd** [comment]
Yep makes sense ! (I missed the previous message sorry) WNBOI is just a library so this would be a feature for another mod. I can throw something together for it when I have time though.

**Comment by Yarden-zamir** [comment]
Ah I see. I would love that yea
